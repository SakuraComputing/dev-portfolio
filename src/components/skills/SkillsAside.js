import React from 'react';
import Skills from './Skills';

class SkillsAside extends React.Component {

    render() {

        const { skillsaside, skillsummary } = this.props;

        let skillItems;

        if(skillsaside.length > 0) {
            skillItems = skillsaside.map(skill => (
                <Skills key={skill.id} skill={skill}/>
            ))
        } else {
            skillItems = <h4>No skillsaside found....... </h4>
        }

        return (
            <aside className="skills aside section">
                <div className="section-inner">
                    <h2 className="heading">Skills</h2>
                    <div className="content">
                        <p className="intro">
                            {skillsummary}
                        </p>
                        <div className="skillset">
                            {skillItems}
                        </div>
                    </div>
                </div>
            </aside>
        );
    }
}
export default SkillsAside;


