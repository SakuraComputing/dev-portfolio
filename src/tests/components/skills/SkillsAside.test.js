import React from 'react';
import { shallow } from 'enzyme';
import SkillsAside from '../../../components/skills/SkillsAside';

describe('SkillsAside Section', () => {
    it('should render correctly', () => {
        const skillsaside =
          [
            {
                "id": 1,
                "title": "CSS",
                "level": "Expert",
                "Bar": 80
            },
            {
                "id": 2,
                "title": "Javascript",
                "level": "Expert",
                "Bar": 81
            }
          ]
        ;
        const wrapper = shallow(<SkillsAside skillsaside={skillsaside}/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('should render no skills found if none passed in', () => {
        const skillsaside = [];
        const wrapper = shallow(<SkillsAside skillsaside={skillsaside}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
