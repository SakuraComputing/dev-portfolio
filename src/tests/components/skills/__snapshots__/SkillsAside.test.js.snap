// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillsAside Section should render correctly 1`] = `
<aside
  className="skills aside section"
>
  <div
    className="section-inner"
  >
    <h2
      className="heading"
    >
      Skills
    </h2>
    <div
      className="content"
    >
      <p
        className="intro"
      />
      <div
        className="skillset"
      >
        <Skills
          key="1"
          skill={
            Object {
              "Bar": 80,
              "id": 1,
              "level": "Expert",
              "title": "CSS",
            }
          }
        />
        <Skills
          key="2"
          skill={
            Object {
              "Bar": 81,
              "id": 2,
              "level": "Expert",
              "title": "Javascript",
            }
          }
        />
      </div>
    </div>
  </div>
</aside>
`;
